### ISSUE TITLE
Advanced Search & Top Rated returns 0.   Private smartlists doesn't show up on DSub

### ISSUE THREAD

**Issue by PsyZeus** [ampache/ampache#3072]
### Describe the bug

After updating to 5.1.0 i have 3 issues so far.

1) the Top rated page "/stats.php?action=highest"
returns "Item Count: 0" to everything

2) the random search "/random.php?action=advanced"
no matter what filter is set returns also 0 results every time

3) Private smartlists doesn't show up on DSub

## Environment

* Ampache version: 5.1.0_all_php8
* Web server + version: Apache 2.4.38
* PHP: 8.0.12
* Server operating system: debian 10

## Log errors

for 1)
(Ampache\Module\System\Dba) -> Error_query SQL: SELECT MIN(`rating`.`object_id`) as `id`, ROUND(AVG(`rating`), 2) AS `rating`, COUNT(DISTINCT(`user`)) AS `count` FROM `rating` WHERE `object_type` = ? GROUP BY `rating`.`object_id` ORDER BY `rating` DESC, `count` DESC, `id` DESC  LIMIT 0, 50

(Ampache\Module\System\Dba) -> Error_query MSG: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '? GROUP BY `rating`.`object_id` ORDER BY `rating` DESC, `count` DESC, `id` DE...' at line 1

for 2)
(Ampache\Module\Application\ApplicationRunner) -> Found handler "Ampache\Module\Application\Random\GetAdvancedAction" for action "get_advanced"

(log.lib) -> [Runtime Error] Undefined array key "type" in file /var/www/html/ampachetest/src/Repository/Model/Search.php(851)

for 3)
(log.lib) -> [Runtime Error] Undefined array key "username" in file /var/www/html/ampachetest/src/Module/Api/Subsonic_Api.php(986)

**Comment by PsyZeus** [comment]
You rocked once again dude!! :D
1 and 3 seems to work awesome after the fixes, i will test some more tomorrow tho just to be sure.

for 2 indeed it needs more work as i still get 0 results and log doesn't seem to report anything after the fixes


log

2021-10-29T21:45:05+03:00 [ampache] (Ampache\Module\System\Dba) -> Database connection...
2021-10-29T21:45:05+03:00 [ampachetest] (Ampache\Module\System\Session) -> 6jffh6c504mi9b6k7rm5ft3bag has been extended to Fri, 29 Oct 2021 22:45:05 +0300 extension length 3600
2021-10-29T21:45:05+03:00 [ampachetest] (Ampache\Module\Application\ApplicationRunner) -> Found handler "Ampache\Module\Application\Random\GetAdvancedAction" for action "get_advanced"
2021-10-29T21:45:05+03:00 [ampachetest] (Ampache\Module\System\Session) -> Session created: 869d172d62a163e5694130bdded52f37
2021-10-29T21:45:05+03:00 [ampachetest] (Ampache\Repository\Model\Browse) -> Show objects called for type {song}
2021-10-29T21:45:05+03:00 [ampachetest] (Ampache\Module\System\Session) -> Writing to 6jffh6c504mi9b6k7rm5ft3bag with expiration 1635536705
2021-10-29T21:45:06+03:00 [ampache] (Ampache\Module\System\Dba) -> Database connection...
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\System\Session) -> 6jffh6c504mi9b6k7rm5ft3bag has been extended to Fri, 29 Oct 2021 22:45:06 +0300 extension length 3600
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\Application\ApplicationRunner) -> Found handler "Ampache\Module\Application\SearchData\ShowAction" for action "show"
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\System\Session) -> Writing to 6jffh6c504mi9b6k7rm5ft3bag with expiration 1635536706
2021-10-29T21:45:06+03:00 [ampache] (Ampache\Module\System\Dba) -> Database connection...
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\System\Session) -> 6jffh6c504mi9b6k7rm5ft3bag has been extended to Fri, 29 Oct 2021 22:45:06 +0300 extension length 3600
2021-10-29T21:45:06+03:00 [ampachetest] (ajax.server) -> Called for page: {browse}
2021-10-29T21:45:06+03:00 [ampachetest] (browse.ajax) -> Called for action: {get_filters}
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\System\Session) -> Writing to 6jffh6c504mi9b6k7rm5ft3bag with expiration 1635536706
2021-10-29T21:45:06+03:00 [ampache] (Ampache\Module\System\Dba) -> Database connection...
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\System\Session) -> 6jffh6c504mi9b6k7rm5ft3bag has been extended to Fri, 29 Oct 2021 22:45:06 +0300 extension length 3600
2021-10-29T21:45:06+03:00 [ampachetest] (Ampache\Module\System\Session) -> Writing to 6jffh6c504mi9b6k7rm5ft3bag with expiration 1635536706
2021-10-29T21:45:09+03:00 [ampache] (Ampache\Module\System\Dba) -> Database connection...
2021-10-29T21:45:09+03:00 [ampachetest] (Ampache\Module\System\Session) -> 6jffh6c504mi9b6k7rm5ft3bag has been extended to Fri, 29 Oct 2021 22:45:09 +0300 extension length 3600
2021-10-29T21:45:09+03:00 [ampachetest] (ajax.server) -> Called for page: {index}
2021-10-29T21:45:10+03:00 [ampachetest] (Ampache\Module\System\Session) -> Writing to 6jffh6c504mi9b6k7rm5ft3bag with expiration 1635536710

**Comment by lachlan-00** [comment]
i think an action or step is missing, i'll be looking again on monday but it shouldn't take long once i've got time to check it out

**Comment by PsyZeus** [comment]
Heh I'm sure about that! :P
Have a nice weekend m8

**Comment by lachlan-00** [comment]
monday is the same as thursday right? working on this now

**Comment by lachlan-00** [comment]
These should all be fixed now
2 is fixed with aca63d4b1, a568e6a86, aefce1d11 and 6fda95466

The action and browse was missing, it's interesting that it even worked in php7.4 :/

**Comment by PsyZeus** [comment]
> 
> 
> monday is the same as thursday right?

hahahahaha :D we all know mondays are tough and sometimes they can last until friday or saturday so YES its the same!



> 
> 
> These should all be fixed now 2 is fixed with [aca63d4](https://github.com/ampache/ampache/commit/aca63d4b1de318a610252d8418ddf260ea1ebf85), [a568e6a](https://github.com/ampache/ampache/commit/a568e6a86c4c3cb09ba1a30d585b7cba8a7dfd3b), [aefce1d](https://github.com/ampache/ampache/commit/aefce1d11ba4ab72c280021fe00be7d2cb447ff6) and [6fda954](https://github.com/ampache/ampache/commit/6fda95466636fd5891c0b49db19cdea347405fae)

wooooo yeaa fixed for sure dude as i just completed a series of tests with various filters and all seemed to work as expected.

THANK YOU ONCE MORE!!!!!!!!!

and now for the love of music some more shuffled music!!
c u around m8

> 
> The action and browse was missing, it's interesting that it even worked in php7.4 :/

![i-dont-know](https://user-images.githubusercontent.com/26334546/140387658-5bb0fc9f-7510-4bf7-bb94-525a245636a1.jpg)
